### PR TITLE
Security fix for `semver` vulnerability

### PR DESCRIPTION
**Pull request by romainmenke**
> Which issue, if any, is this issue related to?

Closes https://github.com/stylelint/stylelint/issues/5042
Closes https://github.com/stylelint/stylelint/issues/7040

> Is there anything in the PR that needs further explanation?

Not sure how to describe this change within the conventions of the project.
Suggestions welcome.

-----

How do we verify that #5042 is fixed?

Comments within that issue describe that updating `meow` is the solution, but it would be nice to verify this and to have a test that prevents regressions.

----

#7040 :

```
npm audit --production
found 0 vulnerabilities
```

### CODE REVIEW

**Comment by jeddy3** [comment]
> Not sure how to describe this change within the conventions of the project.

Let's lead with the security fix. We can manually add a changeset entry for the windows 10 regression.

**Comment by jeddy3** [comment]
> How do we verify that https://github.com/stylelint/stylelint/issues/5042 is fixed?

Not sure. In the past, for intermittent platform-specific issues we've just waited for users to verify it's working after release. Particularly when the issue was patched in one of our dependencies.

File: lib/cli.mjs
Comment (ybiquitous) [inline lib/cli.mjs:2]: [suggest] We can now remove `use strict` since it's ESM 👍🏼 

```suggestion
```

**Review by ybiquitous — COMMENTED** [review]
@romainmenke I don't realize this solution! Thanks!

I still have two concerns:

1. `meow` requires Node.js >=14.16. Is this a breaking change? That is, should we update `engines.node` to `^14.18.0 || >= 16.0.0`?

```console
$ npm view meow@11.0.0 engines
{ node: '>=14.16' }

$ npx ls-engines@latest
...
┌───────────────────────────────────┬───────────────────────────┐
│ package engines:                  │ dependency graph engines: │
├───────────────────────────────────┼───────────────────────────┤
│ "engines": {                      │ "engines": {              │
│   "node": "^14.13.1 || >= 16.0.0" │   "node": ">= 14.18"      │
│ }                                 │ }                         │
└───────────────────────────────────┴───────────────────────────┘
...
```

2. People will not be able to `require "stylelint/lib/cli.js"`. Should this be acceptable? I guess there may be few people who need such a CJS way... 🤔

**Comment by jeddy3** [comment]
> meow requires Node.js >=14.16. Is this a breaking change? That is, should we update engines.node to ^14.18.0 || >= 16.0.0?

That's a shame. It is breaking (although probably for only a handful of users).

No easier answer here. Major releases are painful for plugin authors, so we only do it once a year.

Shall we move this change to `v16`? It's a security fix, but as we're a dev tool the impact should be low.

> People will not be able to require "stylelint/lib/cli.js". Should this be acceptable?

I believe so, as it's not part of our public API.

**Comment by romainmenke** [comment]
> meow requires Node.js >=14.16. Is this a breaking change? That is, should we update engines.node to ^14.18.0 || >= 16.0.0?

`meow` version 11 requires these node versions, but version 10 should be fine.
I did get a typescript warning on 10, but that might be possible to solve.

Version 10 also resolves the security issue.

I will look into this shortly :)

**Comment by Mouvedia** [comment]
> `meow` version 11 requires these node versions, but version 10 should be fine.

I agree but it needs to be at least version 10.1.0.
cf https://github.com/stylelint/stylelint/issues/5042#issuecomment-1416791902

File: patches/meow++type-fest+1.4.0.patch
Comment (romainmenke) [inline patches/meow++type-fest+1.4.0.patch:12]: The types were incorrect at this version, but that doesn't affect the runtime of the cli.

File: tsconfig.json
Comment (romainmenke) [inline tsconfig.json:29]:
```suggestion
	"exclude": ["**/*.test.js", "**/__tests__/"]
```

**Comment by ybiquitous** [comment]
I also agree with updating to `meow@10` instead of `meow@11`. 👍🏼 

For your information:

```console
$ npm view meow@^10.1.0 engines
meow@10.1.0 { node: '>=12.17' }
meow@10.1.1 { node: '>=12.17' }
meow@10.1.2 { node: '^12.20.0 || ^14.13.1 || >=16.0.0' }
meow@10.1.3 { node: '^12.20.0 || ^14.13.1 || >=16.0.0' }
meow@10.1.4 { node: '^12.20.0 || ^14.13.1 || >=16.0.0' }
meow@10.1.5 { node: '^12.20.0 || ^14.13.1 || >=16.0.0' }
```

Perhaps, it may be better to update the latest version of `meow@^10.1.0` like this:

```diff
- "meow": "^10.1.0",
+ "meow": "^10.1.5",
```

**Comment by ybiquitous** [comment]
In addition, here's a result of `ls-engines`:

```console
$ npx ls-engines@latest
...
┌───────────────────────────────────┬───────────────────────────┐
│ package engines:                  │ dependency graph engines: │
├───────────────────────────────────┼───────────────────────────┤
│ "engines": {                      │ "engines": {              │
│   "node": "^14.13.1 || >= 16.0.0" │   "node": ">= 14.18"      │
│ }                                 │ }                         │
└───────────────────────────────────┴───────────────────────────┘
...
Your “engines” field does not exactly match your dependency graph‘s requirements!
...
```

`ls-engines` still complains as above, but I think `meow@^10.1.0` is acceptable because `meow` itself requires `node: '^12.20.0 || ^14.13.1 || >=16.0.0'`. 👍🏼

File: .changeset/twenty-camels-promise.md
Comment (ybiquitous) [inline .changeset/twenty-camels-promise.md:5]: [note] Our maintainer guide now doesn't mention the `Security:` prefix. This doesn't block this PR, but I'm a bit curious. 😅 

https://github.com/stylelint/stylelint/blob/a42f95582c2a928122fc1e59f9f8acc078fb9b91/docs/maintainer-guide/pull-requests.md?plain=1#L22-L23

File: package.json
Comment (ybiquitous) [inline package.json:158]: [suggest] This is a minor suggestion (also `package-lock.json`), but you can ignore it.

```suggestion
    "meow": "^10.1.5",
```

**Review by ybiquitous — APPROVED** [review]
Thank you for the excellent fix. LGTM 👍🏼

**Comment by ybiquitous** [comment]
I think this PR needs one more approvement.

**Comment by romainmenke** [comment]
> `ls-engines` still complains as above

Seems to be because of `supports-hyperlinks`, which is a direct dependency.

```json5
"node_modules/supports-hyperlinks": {
	"version": "3.0.0",
	// ...
	"engines": {
		"node": ">=14.18"
	}
},
```

Luckily not a sub dependency of `meow` v10.x :)

File: .changeset/twenty-camels-promise.md
Comment (jeddy3) [reply .changeset/twenty-camels-promise.md:5]: It's an oversight. It should list all the ones from https://keepachangelog.com/en/1.0.0/. I'll open a PR.

**Review by jeddy3 — APPROVED** [review]
LGTM, thank you very much.

**Pull request merged**
